Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -377,6 +379,26 @@ void AppendPipeline::handleNeedContextSyncMessage(GstMessage* message)
         return;
 
     if (!g_strcmp0(contextType, "drm-preferred-decryption-system-id")) {
+#if USE(OPENCDM)
+        std::pair<Vector<GRefPtr<GstEvent>>, Vector<String>> streamEncryptionInformation = GStreamerEMEUtilities::extractEventsAndSystemsFromMessage(message);
+        for (auto& event : streamEncryptionInformation.first) {
+            const char* eventKeySystemId = nullptr;
+            GstBuffer* data = nullptr;
+            gst_event_parse_protection(event.get(), &eventKeySystemId, &data, nullptr);
+
+            GstMapInfo mapInfo;
+            if (!gst_buffer_map(data, &mapInfo, GST_MAP_READ)) {
+                GST_WARNING("cannot map %s protection data", eventKeySystemId);
+                break;
+            }
+
+            m_initData.append(mapInfo.data, mapInfo.size);
+            if (streamEncryptionInformation.second.contains(eventKeySystemId))
```

Comment:
Do you need a mapping for events here? One pipeline, one init data, one event seqnum, right? Just keep that.

Reply:
As I mentioned for regular flow, the MSE flow also events for all supported protection system. If the content support WV, PR and CK. It will get three events. so that we added this map to get exact event sequence number based on protection system selected